### ISSUE TITLE
Add a lint for `panic!` being used

### ISSUE THREAD

**Issue by sgrif** [rust-lang/rust-clippy#2297]
Given that there's a lint for `Result::unwrap` and `Option::unwrap` being used, I was surprised that this wasn't present.

**Comment by y21** [comment]
This lint exists today (along with lints for all the other kinds of `panic`-like macros).
```rs
#[warn(clippy::panic)]
fn f() {
    panic!();
}
```
```
warning: `panic` should not be present in production code
 --> src/main.rs:3:5
  |
3 |     panic!();
  |     ^^^^^^^^
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#panic
```
